variables.tf

```diff
@@ -91,17 +91,29 @@ variable "runtime" {
     condition = var.runtime == null ? true : contains([
       "nodejs", "nodejs4.3", "nodejs6.10",
       "nodejs8.10", "nodejs10.x", "nodejs12.x",
-      "nodejs14.x", "java8", "java8.al2",
+      "nodejs14.x", "nodejs16.x", "java8", "java8.al2",
       "java11", "python2.7", "python3.6",
       "python3.7", "python3.8", "python3.9",
       "dotnetcore1.0", "dotnetcore2.0", "dotnetcore2.1",
-      "dotnetcore3.1", "nodejs4.3-edge", "go1.x",
+      "dotnetcore3.1", "dotnet6",
+      "nodejs4.3-edge", "go1.x",
       "ruby2.5", "ruby2.7", "provided", "provided.al2"
     ], var.runtime)
     error_message = "Identifier of the function's runtime must be supported by AWS Lambda."
   }
 }
 
+variable "architecture" {
+  description = "Instruction set architecture for your Lambda function. Valid values are 'x86_64' and 'arm64'."
+  type        = string
+  default     = "x86_64"
+
+  validation {
+    condition     = var.architecture == null ? true : contains(["x86_64", "arm64"], var.architecture)
```

Comment:
condition should be 'contains(["x86_64", "arm64"], var.architecture)' because default value is not null

Reply:
thx - took variable "runtime" as pattern.
leaves me with the question, why we have:
variable "runtime" {
  description = "Identifier of the function's runtime. See Runtimes for valid values."
  type        = string
  default     = null

This makes no sens so me - you must specify a runtime